### DISCUSSION TITLE
HanthushWellModel to simulate infiltration

### DISCUSSION THREAD

**Discussion by LovisaBostrom** [pastas/pastas#367] [Q&A] [answered]
Hi,

I am using the HantushWellModel to simulate the effect of two infiltration wells on an observation well in an area. From my understanding the dependent time series (in this case infiltration) needs to have regular time steps during the entire time period that I am trying to solve my model for. 

So for example: I am solving my model for a two year time period, during this period I have scarce  infiltration measurements. I solved this be interpolating measurements but also by adding zero values during the time period where I know the infiltration was not ongoing. However, when I solve my model the results are showing an impact from the infiltration during a time period where there was no infiltration (zero values in the time series).

What is the reason for this? Should the dependent time series be handled differently to avoid this issue?

Thanks in advance.

**Comment by dbrakenhoff** [comment]
Hi @LovisaBostrom,

That sounds odd, but it's hard to tell without seeing the code/results. Can you share your code? And data if possible? I'll take a look and see if I can figure out what is going on.

**Reply by LovisaBostrom** [reply]
Hi,

So this figure shows the time series I have used for the infiltration with a daily frequency.
![image](https://user-images.githubusercontent.com/82801075/151195946-527612d1-ab76-4f35-ae66-8f841c6f4ab0.png)

(Infiltration_2 stopped at the end of 2020 and infiltration_1 has been shut of for the most part from the end of 2020 except for a few days in september/october of 2021. 

I also have leakage measurents and recharge with a daily frequency. 

And this is how I set up the model:


```
ob_func=ps.LmfitSolve 
rfun=ps.Gamma 
repo=False 
rch=ps.rch.Linear()
use_of_noise=True
date_max='2022-01-14' 
date_min= '2018-01-01'

ml_3=ps.Model(Ho, name=WELL)

sm3 = ps.RechargeModel(rain, evap, rfunc=rfun, recharge=rch, name="rch3")

ml_3.add_stressmodel(sm3)

leakage_stress = ps.WellModel(leakage_list, rfunc=ps.HantushWellModel, name="leakage3", distances=dist1, settings="well")
ml_3.add_stressmodel(leakage_stress)

infiltration_stress = ps.WellModel(infiltration_list, rfunc=ps.HantushWellModel, name="infiltration3", distances=dist2, settings="well",  up=True)
ml_3.add_stressmodel(infiltration_stress)

ml_3.solve(report=repo, solver=ob_func, noise=use_of_noise, tmin=date_min, tmax=date_max)
```
I then use ml_3.get_contributions() to plot the contribution each stress has on the simulation. These are my results (here also included the impact from the leakage).

![image](https://user-images.githubusercontent.com/82801075/151198609-9158708f-2b9c-4a4f-8bce-a1cab174b5f9.png)

As you can see infiltration_2 keep increasing towards the beginning of 2022 even though there are zero values in the time series since the end of 2020. This seems unlikely in my opinion and also the fact that the impact is estimated to a change in gw-level of 0.2 m.

I have considered the step response as answered below but it seems like a very delayed response. It should also be added that the observation well is loctated pretty far away from the infiltration wells approximately 600-900 m.

Not sure if this makes it any clearer..

**Comment by mbakker7** [comment, marked as answer]
It is common to see an effect of infiltration (or any other stress) for
quite some time after the stress has stopped (equals zero). That is the
memory of the system. The effect of a stress equals zeros X days after the
stress stopped, where X is the memory. You can estimate the memory by
looking a the step response to see how long it takes before the step
response reaches a plateau.

**Reply by LovisaBostrom** [reply]
Hi,

That makes sense however, I get a very high response up to a year after the infiltration has been shut of, and also an response that keeps increasing. Using ml.get_response_tmax('infiltration', cutoff=0.99) i get 1576 (days I presume) which is of course more than one year.

**Reply by LovisaBostrom** [reply]
Hi again @mbakker7,  

I have an additional question regarding simulating the response due to groundwater infiltration. Would you say that that the Pastas model is suitable for simulating this? Having tried the different response functions that are available I am getting results that I consider to be highly unreasonable. 

Perhaps it is due to my lack of knowledge of setting up the model or that I have inaccurate stresses. However, I was wondering if there are examples when Pastas has been used succesfully for the purpose of simulating groundwater infiltration?

Thanks in advance.

**Comment by mbakker7** [comment]
Lovisa,

Just to be sure: You use the term 'groundwater infiltration'. Do you mean
injection by a well? Or do you mean recharge from rainfall and evaporation.

It is very difficult to assess your model from what you have provided. Can
you make plots.results figure?

MB

**Reply by LovisaBostrom** [reply]
Yes should have clarified, I am reffering to injection into a well. To give som context: I am trying to study the effect from the construction of a tunnel and ultimately the leakage into the tunnel on the groundwater levels in an area. There has been ongoing infiltration into two groundwater wells (both rock and a well in the lower aquifer) in the area to counteract the effect from the tunnel. So I also want to study the effect the two infiltration wells have had on the groundwater levels. 

These are the stresses I am using:
![image](https://user-images.githubusercontent.com/82801075/151546952-55467b68-8523-4128-8719-5d35469e09ab.png)


I have made sure that all of them have a daily frequency (except for the head observations) with zero values where there was no on going infiltration. 

I am using the HantushWellModel to both simulate the effect from tunnel leakage and infiltration:
```
# "settings" 
ob_func=ps.LmfitSolve #solver #LeastSquares
rfun=ps.Gamma #responsefunction
NO=False #Noise
repo=False #report
rch=ps.rch.Linear()#FlexModel()
use_of_noise=False
date_max='2022-01-14'
date_min= '2018-01-01'

# Model with only groundwater recharge
#head observations
ml_1=ps.Model(Ho, name=WELL)
# creating a stress model for the recharge
sm = ps.RechargeModel(rain, evap, rfunc=rfun, recharge=rch, name="rch1") 
#Adding the stressmodel to the model
ml_1.add_stressmodel(sm) 
#Solving the model
ml_1.solve(report=repo, solver=ob_func, noise=use_of_noise,  tmin=date_min, tmax=date_max)
ml_1.plot()

#Model with tunnel leakage
ml_2=ps.Model(Ho, name=WELL)

sm2 = ps.RechargeModel(rain, evap, rfunc=rfun, recharge=rch, name="rch2")
ml_2.add_stressmodel(sm2)

inläckage_stress = ps.WellModel(inläckage, rfunc=ps.HantushWellModel, name="inläckage", distances=avs_inläckage, settings="well")
ml_2.add_stressmodel(inläckage_stress)

ml_2.solve(report=repo, solver=ob_func, noise=use_of_noise, tmin=date_min, tmax=date_max)
ml_2.plot()

#Model with leakage and infiltration
ml_4=ps.Model(Ho, name=WELL)

sm4 = ps.RechargeModel(rain, evap, rfunc=rfun, recharge=rch, name="rch4")
ml_4.add_stressmodel(sm4)

inläckage_stress = ps.WellModel(inläckage, rfunc=ps.HantushWellModel, name="inläckage4", distances=avs_inläckage, settings="well")
ml_4.add_stressmodel(inläckage_stress)

infiltration_stress = ps.WellModel(infiltration_list, rfunc=ps.HantushWellModel, name="infiltration4", distances=avs_inf, settings="well",  up=True)
ml_4.add_stressmodel(infiltration_stress)

ml_4.solve(report=repo, solver=ob_func, noise=use_of_noise, tmin=date_min, tmax=date_max)
ml_4.plot()
```
My results:
![image](https://user-images.githubusercontent.com/82801075/151547100-e9b55a0d-e817-4f7a-8236-2ed45f8165f8.png)

Analysis of the model containing tunnel leakage and infiltration:
![image](https://user-images.githubusercontent.com/82801075/151547052-25eda705-64e9-4906-aa41-a8b97403885b.png)

I then do simulations based on the model with both tunnel leakage and infiltration and quantifying the impact each stress has had:
![image](https://user-images.githubusercontent.com/82801075/151547346-726d35fe-912e-4398-9664-b78eb6e3a95c.png)

![image](https://user-images.githubusercontent.com/82801075/151548376-cdb1285e-d718-4fb7-8f96-079cc0102003.png)

I just find it unlikely that groundwater infiltration would have an effect up to more than one and a half years after the infiltration has been shut of (infiltration in the rock aquifer). Especially since the infiltration wells are located pretty far away 600 - 900 m. So that is why I was wondering if any of the repsonse functions are applicable for this scenario.
